File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.py
def solution(keymap, targets):
    dic = {}
    for key in keymap:
        for idx, val in enumerate(key,1):
            if val in dic.keys() and dic[val] < idx:
                continue
            dic[val] = idx
    ans = []
    
    for target in targets:
        count=0
        for c in target:
            if c in dic.keys():
                count+=dic[c]
            else:
                ans.append(-1)
                count = -1
                break
        if count != -1:
            ans.append(count)
    return ans

File: 프로그래머스/unrated/160586. 대충 만든 자판/대충 만든 자판.js
function solution(keymap, targets) {
    const press_key = {}, answer = []
    
    for(keys of keymap){
        for(let i=0; i < keys.length; i++){
            if(press_key[keys[i]] === undefined){
                press_key[keys[i]] = i+1
            } else{
                if(press_key[keys[i]] > i+1) press_key[keys[i]] = i+1;
            }
        }
    }
    
    for(target of targets){
        let num = 0
        for(c of target){
            if(press_key[c] === undefined){
                num=-1
                break
            }
            num+=press_key[c]
        }
        answer.push(num)
    }
    
    return answer
}